### PR TITLE
Add usermanagement for PostgreSQL

### PR DESCRIPTION
**Pull request by Kidswiss**
We have to enable Crossplane's usages in order for this feature to work correctly.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: component/class/defaults.yml [hunk 1]
```diff
@@ -45,7 +49,7 @@ parameters:
       appcat:
         registry: ghcr.io
         repository: vshn/appcat
-        tag: v4.71.2
+        tag: add/user_management
```
Comment (Kidswiss) [inline component/class/defaults.yml:52]: Will be adjusted once there's a tag

**Review by wejdross — APPROVED** [review]

File: component/class/defaults.yml [hunk 2]
```diff
@@ -402,6 +406,16 @@ parameters:
         providerConfigRefs: []
         additionalProviderConfigs: []
 
+      sql:
+        enabled: false
```
Comment (zugao) [inline component/class/defaults.yml:410]: maybe we should not exposed this parameter? `provider-sql` must be available for services that expose user management. What would happen if the sql provider is disabled meanwhile the service postgres allows user management?

File: component/component/provider.jsonnet [hunk 3]
```diff
@@ -383,4 +393,47 @@ local runtimeConfigRef(name) = {
       role,
       rolebinding,
     ],
+  [if params.providers.sql.enabled || params.services.vshn.postgres.enabled then '10_provider_sql']:
```
Comment (zugao) [inline component/component/provider.jsonnet:396]: I would not use `params.providers.sql.enabled` all together. Just enabled provider sql for any service that needs it.

**Review by zugao — COMMENTED** [review]
Some things to consider

File: component/component/provider.jsonnet (hunk 3 above)
Comment (Kidswiss) [reply component/component/provider.jsonnet:396]: Fine by me

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**